src/app/cli/src/init/coda_run.ml

```diff
@@ -64,6 +64,61 @@ let get_lite_chain :
       let proof = Lite_compat.proof proof in
       {Lite_base.Lite_chain.proof; ledger; protocol_state} )
 
+let get_current_fork_id ~conf_dir ~logger =
+  let current_fork_id_file = conf_dir ^/ "current_fork_id" in
+  let read_fork_id () =
+    let open Stdlib in
+    let inp = open_in current_fork_id_file in
+    let res = input_line inp in
+    close_in inp ; res
+  in
+  let write_fork_id fork_id =
+    let open Stdlib in
+    let outp = open_out current_fork_id_file in
+    output_string outp (fork_id ^ "\n") ;
+    close_out outp
+  in
+  function
+  | None -> (
+    try
+      (* not provided on command line, read from config dir *)
+      let fork_id = read_fork_id () in
+      Logger.info logger ~module_:__MODULE__ ~location:__LOC__
+        "Set current fork ID to $fork_id from config"
+        ~metadata:[("fork_id", `String fork_id)] ;
+      fork_id
+    with Sys_error _ ->
+      Logger.fatal logger ~module_:__MODULE__ ~location:__LOC__
+        "No current fork ID provided on command line, and none available in \
+         config" ;
+      failwith "No current fork ID available" )
+  | Some fork_id -> (
+    try
+      (* it's an error if the command line value disagrees with the value in the config *)
+      let config_fork_id = read_fork_id () in
+      if String.equal config_fork_id fork_id then (
+        Logger.info logger ~module_:__MODULE__ ~location:__LOC__
+          "Using current fork ID $fork_id from command line, which matches \
+           the one in the config"
+          ~metadata:[("fork_id", `String fork_id)] ;
+        config_fork_id )
+      else (
+        Logger.fatal logger ~module_:__MODULE__ ~location:__LOC__
+          "Current fork ID $fork_id from the command line disagrees with \
+           $config_fork_id from the config"
+          ~metadata:
+            [ ("fork_id", `String fork_id)
+            ; ("config_fork_id", `String config_fork_id) ] ;
+        failwith
+          "Current fork ID from command line disagrees with fork ID in config" )
```

Comment:
We should probably provide a helpful error message to the user that tells them to delete their coda config and try again if they want to switch fork IDs. Alternatively, we could provide a `--fresh` command that automates this for them and encourage them to add that flag to delete their persisted data, but that might be too destructive.

Reply:
added to the existing error message, suggesting removing config